File: Java/ADReNA_API/src/ADReNA_API/Backpropagation/BackpropagationLayer.java
package ADReNA_API.Backpropagation;

import java.util.ArrayList;

public class BackpropagationLayer {

    public ArrayList<BackpropagationNeuron> neurons;

    public BackpropagationLayer()
    {
        neurons = new ArrayList<BackpropagationNeuron>();
    }

    public BackpropagationLayer(int layerSize)
    {
        neurons = new ArrayList<BackpropagationNeuron>();
        for (int x = 0; x < layerSize; x++)
            neurons.add(new BackpropagationNeuron());
    }

    public BackpropagationLayer(int layerSize, BackpropagationLayer fatherLayer)
    {
        neurons = new ArrayList<BackpropagationNeuron>();
        for (int x = 0; x < layerSize; x++)
        {
            BackpropagationNeuron neuron = new BackpropagationNeuron();
            neurons.add(neuron);

            for (BackpropagationNeuron fatherNeuron : fatherLayer.neurons)
            {
                neuron.listConnection.add(new BackpropagationConnection(fatherNeuron));
            }
        }
    }

}

File: NET/ExemploKohonem/ADReNA_API/NeuralNetwork/Backpropagation/BackpropagationLayer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ADReNA_API.NeuralNetwork
{
    public class BackpropagationLayer
    {
        #region Attributes
        public List<BackpropagationNeuron> neurons;
        #endregion

        #region private Methods
        #endregion

        #region public Methods
        public BackpropagationLayer()
        {
            neurons = new List<BackpropagationNeuron>();
        }

        public BackpropagationLayer(int layerSize)
        {
            neurons = new List<BackpropagationNeuron>();
            for (int x = 0; x < layerSize; x++)
                neurons.Add(new BackpropagationNeuron());
        }

        public BackpropagationLayer(int layerSize, BackpropagationLayer fatherLayer)
        {
            neurons = new List<BackpropagationNeuron>();
            for (int x = 0; x < layerSize; x++)
            {
                BackpropagationNeuron neuron = new BackpropagationNeuron();
                neurons.Add(neuron);

                foreach (BackpropagationNeuron fatherNeuron in fatherLayer.neurons)
                {
                    neuron.listConnection.Add(new BackpropagationConnection(fatherNeuron));
                }
            }
        }
        #endregion
    }
}
